milestone-3/frontend/src/components/recipePage.tsx

```diff
@@ -0,0 +1,102 @@
+import  { ChangeEvent, useEffect, useState } from 'react';
+
+
+interface Recipe {
+    name: string;
+    image: string;
+    description: string;
+    ingredients: string [];
+    instructions: string [];
+}
+
+function RecipePage(props: Recipe) {
+  
+
+  const [newIngredient, setNewIngredient] = useState('');
+  const [allIngredients, setAllIngredients] = useState(props.ingredients);
+  const [newInstruction, setNewInstruction] = useState('');
+  const [allInstructions, setAllInstructions] = useState(props.instructions);
+  const addIngredient = () => {
+
+    fetch(`http://localhost:3001/recipe/${props.name}/ingredient`, {
+        method: 'PUT',
+        headers: {
+            'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({
+            ingredient: newIngredient
+        })
+        
+    })
+    .then((res) => console.log(res));
+    setAllIngredients([...allIngredients, newIngredient]);
+
```

Comment:
it could be nice to include a line here to reset the newIngredient field to the empty string so that it resets and allows people to add several ingredients at once. This would look like setNewIngredient(''), after your setAllIngredients line. Check out this behavior when you get a chance.